Data/SessionService.cs

```diff
@@ -10,19 +10,65 @@ public sealed class SessionService
     public SessionService(IDbConnection dbConnection) =>
         _dbConnection = dbConnection;
 
-    public IReadOnlyList<Session> GetSessions() =>
-        _dbConnection.Query<Session, Author, Session>(
-            @"SELECT s.id AS sessionId, s.when, a.id AS authorId, a.name, a.surname 
-                FROM session s
-                JOIN author a
-                    ON s.chair_id = a.id",
-            (s, a) =>
+    public IReadOnlyList<Session> GetSessions() {
+        var sessionDictionary = new Dictionary<int, Session>();
+
+        var lectureDictionary = new Dictionary<int, Lecture>();
+
+        var sessions = _dbConnection.Query<Session, Author, Lecture, Author, Paper, Author, Session>(
+            @"SELECT s.id AS sessionId, s.""when"", 
+                     ch.id AS authorId, ch.name, ch.surname, 
+                     l.id AS lectureId, l.""when"",
+                     sp.id AS authorId, sp.name, sp.surname,
+                     p.id AS paperId, p.name, p.classification,
+                     a.id AS authorId, a.name, a.surname
+              FROM ""session"" s
+              JOIN author ch
+                ON s.chair_id = ch.id
+              LEFT JOIN lecture l
+                ON l.session_id = s.id
+              LEFT JOIN paper p
+                ON p.id = l.paper_id
+              LEFT JOIN author sp
+                ON sp.id = l.speaker_id
+              LEFT JOIN paper_author pa
+                ON pa.paper_id = p.id
+              LEFT JOIN author a
+                ON a.id = pa.author_id",
+            (s, ch, l, sp, p, a) =>
             {
-                s.Chair = a;
-                return s;
+                Session? session;
+                Lecture? lecture;
+
+                if (!sessionDictionary.TryGetValue(s.Id, out session)) {
+                    session = s;
+                    sessionDictionary.Add(session.Id, session);
+                }
+
+                if (l is not null && !lectureDictionary.TryGetValue(l.Id, out lecture)) {
+                    lecture = l;
+                    lectureDictionary.Add(lecture.Id, lecture);
+                    p.Authors.Add(a);
+                    l.Paper = p;
+                    l.Speaker = sp;
+                    session.Lectures.Add(l);
+                }
+                else if (l is not null) {
+                    lecture = session.Lectures.Find(x => x.Id == l.Id);
```

Comment:
Ten lookup jest zbędny. Jak jesteśmy w tym miejscu, to `lecture` jest już przypisany na wartość ze słownika `lectureDictionary`. A więc `lecture` przed tą instrukcją i po tej instrukcji będzie dokładnie taki sam.

Reply:
Tak właściwie to wszystko to da się przepisać na:
```csharp
if (l is not null)
{
    if (!lectureDictionary.TryGetValue(l.Id, out lecture))
    {
        lecture = l;
        lectureDictionary.Add(lecture.Id, lecture);
        l.Paper = p;
        l.Speaker = sp;
        session.Lectures.Add(l);
    }

    lecture.Paper!.Authors.Add(a);
}
```

co jest trochę prostsze IMO.